addons/pos_self_order/static/src/app/pages/product_list_page/product_list_page.js

```diff
@@ -85,6 +85,14 @@ export class ProductListPage extends Component {
             },
             () => [this.selfOrder.currentCategory]
         );
+
+        onWillStart(() => {
+            this.onWillStart();
+        });
+    }
+
+    onWillStart() {
+        this.selfOrder.updateCategoryList();
```

Comment:
May I know why is this necessary? The setup of the store already does this calculation.

Reply:
The setup of the self_order_service is going through the updateCaregoryList one time at the initialization of the self service. Here I want to recompute everytime the product_screen is displayed if the categories are still available or not depending on the time of the day

Reply:
The setup of the self_order_service is going through the updateCaregoryList one time at the initialization of the self service. Here I want to recompute everytime the product_screen is displayed if the categories are still available or not depending on the time of the day